pipeline/splitters.go

```diff
@@ -97,6 +102,87 @@ func (t *TokenSplitter) FindRecord(buf []byte) (bytesRead int, record []byte) {
 	return bytesRead, buf[:bytesRead]
 }
 
+type MultilineSplitter struct {
+	delimiter  *regexp.Regexp
+	multiline  *regexp.Regexp
+	captureLen int
+}
+
+type MultilineSplitterConfig struct {
+	Delimiter string
+	Multiline string
+}
+
+func (r *MultilineSplitter) ConfigStruct() interface{} {
+	return &MultilineSplitterConfig{
+		Delimiter: "\n",
+	}
+}
+
+func (r *MultilineSplitter) Init(config interface{}) error {
+	conf := config.(*MultilineSplitterConfig)
+	var err error
+	if r.delimiter, err = regexp.Compile(conf.Delimiter); err != nil {
+		return err
+	}
+	if r.delimiter.NumSubexp() > 1 {
+		return fmt.Errorf("regex must not contain more than one capture group: %s",
+			conf.Delimiter)
+	}
```

Comment:
Hmm, good catch. I think that's a leftover from some code that was in the `RegexSplitter`, which has this restriction. The reason I don't allow you to configure whether or not you can keep the delimiter is that it will always keep all the intervening delimiters–it doesn't make sense otherwise. But I suppose it might still make sense to configure whether or not you keep the last one.